### ISSUE TITLE
[api] duplicated fields for Math constants on JS / Lua

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#6088]
See http://api.haxe.org/Math.html:

![](http://i.imgur.com/q5JefuY.png)

Same for `POSITIVE_INFINITY` and `PI`.

The difference seems to be that on JS / Lua, they're `(get,null)` properties, while they're regular variables on other targets - so they don't get merged?

**Comment by jdonaldson** [comment]
I can follow suit with js here, I thought it seemed like a good practice to not allow modifying platform constants.

**Comment by jdonaldson** [comment]
My guess is we'll clean this up for haxe 4.  @Simn you want this as a feature request?

**Comment by Simn** [comment]
> @Simn you want this as a feature request?

What exactly?

**Comment by jdonaldson** [comment]
Do you want to protect platform constants behind getters, or do you want to expose them directly?  

The getter changes the signature definition for js/lua, which makes them appear separately in the haxe docs.  

This isn't a high priority imho, we can mark for haxe 4 and get to it later.

**Comment by Aurel300** [comment]
Just wanted to ad that this happens with functions with identical signatures as well, e.g.:

http://api.haxe.org/sys/net/Socket.html#select

`Socket` is an extern class, so the function is only defined once, but it is displayed three times for various platforms. I assume the issue in this case has to do with the argument types resolving to different classes for different platforms?

**Comment by jdonaldson** [comment]
Yeah, things are a bit muddy here, but I doubt we fix for Haxe 3.  I'm going to flag this as an enhancement request for now so that we revisit it for Haxe 4.

**Comment by markknol** [comment]
Isn't this dox issue?

**Comment by Simn** [comment]
This is still an issue.
